test/cypress/tests/collaborative-editing/diff/createOperationByStringsDiff.spec.ts

```diff
@@ -0,0 +1,222 @@
+import { OperationType } from '../../../../../src/components/collaborative-editing/diff/types';
+
+describe('getChangesFromString function', () => {
+  describe('User inserts symbols', () => {
+    it('should create Insert operation when a user inserts symbols at the start of the string', () => {
+      const before = 'string';
+      const after = 'Changed string';
+      const expected = [
+        {
+          type: OperationType.Insert,
+          index: 0,
+          data: 'Changed ',
+        },
+      ];
+
+      const result = createOperationByStringsDiff(before, after);
+
+      expect(result).deep.equal(expected);
+    });
+
+    it('should create Insert operation when a user inserts symbols inside the string', () => {
+      const before = 'string';
+      const after = 'stri123ng';
+      const expected = [
+        {
+          type: OperationType.Insert,
+          index: 4,
+          data: '123',
+        },
+      ];
+
+      const result = createOperationByStringsDiff(before, after);
+
+      expect(result).deep.equal(expected);
+    });
+
+    it('should create Insert operation when a user inserts symbols at the end of the string', () => {
+      const before = 'string';
+      const after = 'string is changed';
+      const expected = [
+        {
+          type: OperationType.Insert,
+          index: 6,
+          data: ' is changed',
+        },
+      ];
+
+      const result = createOperationByStringsDiff(before, after);
+
+      expect(result).deep.equal(expected);
+    });
+
+    it('should create Insert operation when a user inserts symbols inside the empty string', () => {
+      const before = '';
+      const after = 'String';
+      const expected = [
+        {
+          type: OperationType.Insert,
+          index: 0,
+          data: 'String',
+        },
+      ];
+
+      const result = createOperationByStringsDiff(before, after);
+
+      expect(result).deep.equal(expected);
+    });
+  });
+
+  describe('User removes symbols', () => {
+    it('should create Remove operation when a user removes symbols at the start of the string', () => {
+      const before = 'string';
+      const after = 'ing';
+      const expected = [
+        {
+          type: OperationType.Remove,
+          index: 0,
+          data: 'str',
+        },
+      ];
+
+      const result = createOperationByStringsDiff(before, after);
+
+      expect(result).deep.equal(expected);
+    });
+
+    it('should create Remove operation when a user removes symbols from the center of the string', () => {
+      const before = 'String is changed';
+      const after = 'String changed';
+      const expected = [
+        {
+          type: OperationType.Remove,
+          index: 7,
+          data: 'is ',
+        },
+      ];
+
+      const result = createOperationByStringsDiff(before, after);
+
+      expect(result).deep.equal(expected);
+    });
+
```

Comment:
what about a case with several differences?

```js
const before = 'String is changed'
const after = 'Number is changed many times'
```

What will return the `createOperationByStringsDiff` method? Do we need to add test cases for that?

Reply:
We decided, that a user can't do several changes. In this case, the algorithm will return remove and insert operations (like the user replaced the first string with the second one.

Reply:
ok, but add a test case for that please

Reply:
Done!